### ISSUE TITLE
Provide a way to use an alternate JSON mapper

### ISSUE THREAD

**Issue by MT-Jacobs** [micronaut-projects/micronaut-elasticsearch#328]
### Feature description

Using the default Micronaut `ObjectMapper` is good and all when setting up Elasticsearch clients:

https://github.com/micronaut-projects/micronaut-elasticsearch/blob/48e0b0c/elasticsearch/src/main/java/io/micronaut/elasticsearch/DefaultElasticsearchClientFactory.java#L83-L88

but let's say our service is using multiple features that rely on using a shared `ObjectMapper`, like the HTTP server _and_ elasticsearch-java.

If you provide any serialization specialization for Elasticsearch, it impacts the HTTP server, and vice-versa. 

As one example, the default `ObjectMapper` that Micronaut provides doesn't serialize dates correctly; passing an `Instant` within a class results in the fields being serialized to an array, which to ES looks like we're passing multiple dates of very low values (2022 in the Unix Epoch is 2,022 seconds after Jan 1 1970). The workaround is to apply a `@JsonFormat` override to _every_ `Instant`:

```
@JsonFormat(shape = JsonFormat.Shape.STRING)
```

If we attempt to use write-dates-as-timestamps as documented here:

https://github.com/micronaut-projects/micronaut-core/issues/240#issuecomment-394004403

this will fix elasticsearch-java's serialization behavior, but it will also change the http-server serialization behavior.

One solution would be to provide an alternate `ObjectMapper` from the default one for micronaut-elasticsearch. Is this a workable idea?

**Comment by graemerocher** [comment]
yeah that is certainly an idea, wonder if we need to add support in Micronaut Serialization for configuring alternative mappers

**Comment by MT-Jacobs** [comment]
Not sure how I missed this before, but I bet we could use `@JacksonFeatures` to facilitate this the same way that declarative HTTP clients do:

https://docs.micronaut.io/3.7.4/guide/#clientJackson

Will definitely require a code change here though.
